ktor-shared/ktor-sse/common/src/io/ktor/sse/ServerSentEvent.kt

```diff
@@ -15,16 +15,22 @@ import io.ktor.utils.io.*
  *  @property retry reconnection time, in milliseconds to wait before reconnecting.
  *  @property comments comment lines starting with a ':' character.
  */
-public class ServerSentEvent(
-    public val data: String? = null,
+public class ServerSentEvent<T>(
+    public val data: T? = null,
     public val event: String? = null,
     public val id: String? = null,
     public val retry: Long? = null,
     public val comments: String? = null
 ) {
```

Comment:
Would be nice if this was a data class, so instead of having:

```
incoming.collect { person ->
   println(person.data)
}
```

We could have

```
incoming.collect { (person) ->
    println(person)
}
```